### PR TITLE
[FIREFLY-1079] Update Plotly to v2.18.0

### PR DESCRIPTION
**Pull request by tgoldina**
Ticket: https://jira.ipac.caltech.edu/browse/FIREFLY-1079

Test deployment: https://fireflydev.ipac.caltech.edu/firefly-1079-plotly2-18/firefly/

- Updated Plotly from v1.58.4 to v 2.18.0
- Plotly now supports persistent selections: additional selection areas can be added with Shift-MouseDrag. However Firefly filtering from a chart currently works for a single rectangular selection. If multiple areas are selected, Firefly will issue a warning, but preserve the selections layer. Selecting points in multiple areas is allowed. (The points then can be filtered from the table.) 
- Removing selections layer now requires removing `layout.selections`.

### CODE REVIEW

File: src/firefly/js/charts/ui/PlotlyChartArea.jsx [hunk 1]
```diff
@@ -260,15 +263,21 @@ function onSelect(chartId) {
                 } else {
                     dispatchChartUpdate({
                         chartId,
-                        changes: {'selection': {points, range: {x: [xMin, xMax], y: [yMin, yMax]}}}
+                        changes: {
+                            'selection': {
```
Comment (robyww) [inline src/firefly/js/charts/ui/PlotlyChartArea.jsx:267]: you don't need the single quotes just `selection : {`

File: src/firefly/js/charts/ui/PlotlyChartArea.jsx [hunk 2]
```diff
@@ -239,6 +239,9 @@ function onClick(chartId) {
 function onSelect(chartId) {
     return (evData) => {
         if (evData) {
+            // multiple selections are not supported
+            const selections = get(evData, 'selections', [])
```
Comment (robyww) [inline src/firefly/js/charts/ui/PlotlyChartArea.jsx:243]: we are trying to get away from `get`, instead use the new syntax.
```js
const selections = evData?.selections ?? []
```

**Review by robyww — COMMENTED** [review]
Just took a quick look at the code and left a couple of small comments. I will let @loitly do the real review.

Thanks for the work.

**Review by loitly — APPROVED** [review]
Changes look good.

**Pull request merged**